src/seedu/addressbook/data/tag/Tagging.java

```diff
@@ -0,0 +1,32 @@
+package seedu.addressbook.data.tag;
+
+import seedu.addressbook.data.person.Person;
+
+
+
+public class Tagging {
+
+    private TaggingAction action ;
+    private Person person;
+    private Tag tag;
+
+    public Tagging(Person person, Tag tag, TaggingAction action ){
+        this.person = person;
+        this.tag = tag;
+        this.action = action;
+    }
+
+    public String toString(){
+        String actionIndicator = new String();
+
+        if (this.action == TaggingAction.ADDING){
+            actionIndicator = "+";
+        }else if(this.action == TaggingAction.DELETING){
+            actionIndicator = "-";            
+        }
+
+        return actionIndicator + " " + person.getName().toString() + " " + tag.toString();
```

Comment:
We can check if `actionIndicator` has been assigned a character, to avoid the extra preceding whitespace.